### PR TITLE
[Basic] Add SHA256 implementation

### PR DESCRIPTION
**Pull request by aciidb0mb3r**

### CODE REVIEW

File: Tests/BasicTests/SHA2Tests.swift [hunk 1]
```diff
@@ -0,0 +1,43 @@
+/*
```
Comment (ddunbar) [inline Tests/BasicTests/SHA2Tests.swift]: This file should be SHA256Tests, right?

Comment (aciidb0mb3r) [inline Tests/BasicTests/SHA2Tests.swift]: Yep, fixed

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

**Pull request merged**